src/jit/codegenarmarch.cpp

```diff
@@ -1359,6 +1359,11 @@ void CodeGen::genCodeForIndir(GenTreeIndir* tree)
     genConsumeAddress(tree->Addr());
     emit->emitInsLoadStoreOp(ins_Load(targetType), emitTypeSize(tree), targetReg, tree);
     genProduceReg(tree);
+
+    if (tree->gtFlags & GTF_IND_VOLATILE)
+    {
+        instGen_MemoryBarrier();
+    }
```

Comment:
I think using Ld Acquire and Store Release for the LdInd and StInd forms is the first step.  (Block forms look less prevalent and are more complicated)

There are address form and alignment restrictions.  I'll take a look it may still be relatively trivial.